### PR TITLE
Remove isNewAndroidEmbeddingEnabled flag when reading an existing pro…

### PR DESCRIPTION
**Pull request by blasten**
## Description

The flag is still relevant for `flutter create`.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/macos/cocoapods_test.dart [hunk 1]
```diff
@@ -273,7 +273,7 @@ void main() {
         ..writeAsStringSync('Existing release config');
 
       final FlutterProject project = FlutterProject.fromPath('project');
-      await injectPlugins(project);
+      await injectPlugins(project, checkProjects: true);
```
Comment (amirh) [inline packages/flutter_tools/test/general.shard/macos/cocoapods_test.dart]: is this related to this PR?

File: packages/flutter_tools/lib/src/plugins.dart [hunk 2]
```diff
@@ -360,9 +360,6 @@ List<Map<String, dynamic>> _extractPlatformMaps(List<Plugin> plugins, String typ
 /// Returns the version of the Android embedding that the current
 /// [project] is using.
 String _getAndroidEmbeddingVersion(FlutterProject project) {
```
Comment (amirh) [inline packages/flutter_tools/lib/src/plugins.dart]: Unrelated to this PR, just curious - would it be possible / make sense for the logic here and in _getEmbeddingVersion in `platform_plugins.dart` to be the same? (seems like one is deciding based on a manifest key and the other based on Java imports?)

**Review by amirh — APPROVED** [review]
LGTM

File: packages/flutter_tools/test/general.shard/macos/cocoapods_test.dart (hunk 1 above)
Comment (blasten) [reply packages/flutter_tools/test/general.shard/macos/cocoapods_test.dart]: Yep. When this flag is true, `injectPlugins` skips generating the Android plugin registrant since there isn't an `android/` directory in the project under test.

File: packages/flutter_tools/lib/src/plugins.dart (hunk 2 above)
Comment (blasten) [reply packages/flutter_tools/lib/src/plugins.dart]: We don't have the app's main class in pubspec.yaml, so that would require an extra hop. One possibility is that we get the main app class/package from `AndroidManifest.xml` and then check the import in the java class.  That would still require to parse the XML file.
